src/inmanta/server/agentmanager.py

```diff
@@ -243,57 +243,87 @@ async def agent_action(self, env: data.Environment, name: str, action: AgentActi
         if not env.halted and action in {AgentAction.keep_paused_on_resume, AgentAction.unpause_on_resume}:
             raise Forbidden("Cannot set on_resume state of agents when the environment is not halted.")
         if action is AgentAction.pause:
-            await self._pause_agent(env, name)
+            await self._stop_agent(env, name)
         elif action is AgentAction.unpause:
-            await self._unpause_agent(env, name)
+            await self._start_agent(env, name)
         elif action is AgentAction.keep_paused_on_resume:
             await self._set_unpause_on_resume(env, should_be_unpaused_on_resume=False, endpoint=name)
         elif action is AgentAction.unpause_on_resume:
             await self._set_unpause_on_resume(env, should_be_unpaused_on_resume=True, endpoint=name)
         else:
             raise BadRequest(f"Unknown agent action: {action.name}")
 
-    async def _pause_agent(
+    async def _stop_agent(
         self, env: data.Environment, endpoint: Optional[str] = None, connection: Optional[asyncpg.connection.Connection] = None
```

Comment:
Could you describe what is changed here, as I can't make sense of it.  (and this code is extremely sensitive)

Reply:
The only difference is that if the scheduler is used, we call one of its endpoints to perform the action (and we first make sure that the agent client is up and running)